src/store/modules/tag.js

```diff
@@ -0,0 +1,30 @@
+import { fetchTags } from '@/api/tags'
+
+export default {
+  namespaced: true,
+  state: {
+    list: [],
+    loading: false
+  },
+  actions: {
+    fetch ({ commit }, params) {
+      commit('mutateLoading', true)
+      fetchTags(params)
+        .then(result => commit('mutateList', result.data))
+        .finally(() => commit('mutateLoading', false))
+    }
+  },
+  mutations: {
+    mutateList: (state, tags) => {
+      state.list = tags
+    },
+    mutateLoading: (state, loading) => {
+      state.loading = loading
+    }
+  },
+  getters: {
+    getList: (state) => state.list,
+    getLoading: (state) => state.loading,
+    getByName: (state) => (name) => state.list.find(element => element.name === name)
```

Comment:
user can have same name, dont track it by name

Reply:
This applies to tags only